cardano-db-sync/src/Cardano/DbSync/Default.hs

```diff
@@ -85,6 +91,10 @@ applyAndInsertBlockMaybe syncEnv cblk = do
           void $ migrateStakeDistr syncEnv (apOldLedger applyRes)
           insertBlock syncEnv cblk applyRes True tookSnapshot
           liftIO $ setConsistentLevel syncEnv Consistent
+          -- now that we have caught up with the tip of the chain
+          -- we can put the constraints on rewards table
+          lift addRewardTableConstraint
+          lift addEpochStakeTableConstraint
```

Comment:
This rollback should only happens if at least one block is deleted in `rollbackFromBlockNo`